### ISSUE TITLE
Permission denied when copying backup to samba share

### ISSUE THREAD

**Issue by AnonymousRetard** [jcwillox/hass-auto-backup#64]
Not sure if this actually affects anything other than creating ugly error messages in the log every time the scheduled backup runs. But it seems like the chmod_func fails with error "Permission denied" every time a backup has been created and copied to my download_path, which is a mounted samba share.

Is it really necessary to copy all permission bits to the download_path? It seems like if shutil.copyfile would be used instead of shutil.copy then it wouldn't try to run chmod_func and it wouldn't generate any error messages.

See: https://stackoverflow.com/questions/11835833/why-would-shutil-copy-raise-a-permission-exception-when-cp-doesnt

When I try to run chmod or chown on any files on the mounted share as root it doesn't generate any errors but also nothing at all happens. The owner user and group and permissions on all files are controlled by options in /etc/fstab and it's not possible to change on any file or folder on the share.

```
jul 30 02:00:24 htpc hass[1030536]: 2022-07-30 02:00:23 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
jul 30 02:00:24 htpc hass[1030536]: Traceback (most recent call last):
jul 30 02:00:24 htpc hass[1030536]:   File "/etc/homeassistant/custom_components/auto_backup/handlers.py", line 188, in download_backup
jul 30 02:00:24 htpc hass[1030536]:     shutil.copy(backup.path, destination)
jul 30 02:00:24 htpc hass[1030536]:   File "/usr/lib/python3.9/shutil.py", line 428, in copy
jul 30 02:00:24 htpc hass[1030536]:     copymode(src, dst, follow_symlinks=follow_symlinks)
jul 30 02:00:24 htpc hass[1030536]:   File "/usr/lib/python3.9/shutil.py", line 317, in copymode
jul 30 02:00:24 htpc hass[1030536]:     chmod_func(dst, stat.S_IMODE(st.st_mode))
jul 30 02:00:24 htpc hass[1030536]: PermissionError: [Errno 1] Operation not permitted: '/mnt/DroboFS/Shares/Our_stuff/homeassistant-backups/Core_2022_7_7.tar'
```

**Comment by jcwillox** [comment]
Yeah, that makes sense I think I was actually planning to use `shutil.copy2` to preserve metadata as well but had no idea that would cause errors on samba shares. I'd expect it to fail gracefully when the platform doesn't support it.

As you said there's no real reason to copy the permission bits so I'll switch it to `shutil.copyfile`.

**Comment by jcwillox** [comment]
This should be fixed in version [1.2.0](https://github.com/jcwillox/hass-auto-backup/releases/tag/1.2.0) 🎉
